### PR TITLE
Refactoring demo mode

### PR DESCRIPTION
**Pull request by alexjoybc**
# Description

This PR includes the following proposed change(s):

- Adding a new lib to handle demo mode
- Refactoring to add service interface in common
- Up coverage on fee and account service

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Running the app locally

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/backend/Dockerfile [hunk 1]
```diff
@@ -3,22 +3,22 @@
...
 ENV SERVICE_NAME=${SERVICE_NAME}
 
 COPY . .
 
+RUN mvn -B clean install \
+        -P libs
+
+WORKDIR ${SERVICE_NAME}
+
+RUN ls
```
Comment (akroon3r) [inline src/backend/Dockerfile:18]: do we need an ls for debugging?

File: src/backend/libs/efiling-lookup-client/src/test/java/ca/bc/gov/open/jag/efilinglookupclient/CSOLookupServiceImplTest.java [hunk 2]
```diff
@@ -3,54 +3,113 @@
...
+        Mockito
+                .when(lookupFacadeBeanMock.getServiceFee(Mockito.eq(SERVICE_ID_EXCEPTION), Mockito.any()))
+                .thenThrow(new NestedEjbException_Exception("random"));
+
+        sut = new CSOLookupServiceImpl(lookupFacadeBeanMock);
+
     }
 
-    @DisplayName("CASE 1: getServiceFee called with empty serviceId")
+    @DisplayName("Exception: with null service id should throw IllegalArgumentException")
```
Comment (akroon3r) [inline src/backend/libs/efiling-lookup-client/src/test/java/ca/bc/gov/open/jag/efilinglookupclient/CSOLookupServiceImplTest.java:84]: is there a status code associated with this?

File: src/backend/efiling-api/README.md [hunk 3]
```diff
@@ -1,5 +1,24 @@
 # efiling-api
 
+## Maven Profiles
+
+### Openshift
+
+Current image build in openshift runs with `openshift` profile by default.
+This profiles mostly enables splunk logging.
```
Comment (sdevalapurkar-bcgov) [inline src/backend/efiling-api/README.md:8]: change to either:
`This profile mostly enables splunk logging.`
OR
`These profiles mostly enable splunk logging.`

File: src/backend/efiling-api/README.md [hunk 4]
```diff
@@ -1,5 +1,24 @@
...
+
+```bash
+mvn install -P openshift
+```
+
+### Demo
+
+This profiles self isolate the application from any third party dependencies
```
Comment (sdevalapurkar-bcgov) [inline src/backend/efiling-api/README.md:16]: again, change to either:
`This profile self isolates the application from any third party dependencies`
OR
`These profiles self isolate the application from any third party dependencies`

File: src/backend/libs/efiling-demo-starter/src/main/resources/META-INF/spring.factories [hunk 5]
```diff
@@ -0,0 +1,2 @@
+org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
+    ca.bc.gov.open.jag.efiling.demo.AutoConfiguration
```
Comment (sdevalapurkar-bcgov) [inline src/backend/libs/efiling-demo-starter/src/main/resources/META-INF/spring.factories:2]: nit: missing newline

File: src/backend/Dockerfile (hunk 1 above)
Comment (alexjoybc) [reply src/backend/Dockerfile:18]: Thanks!

File: src/backend/efiling-api/README.md (hunk 3 above)
Comment (alexjoybc) [reply src/backend/efiling-api/README.md]: thanks

File: src/backend/libs/efiling-demo-starter/src/main/resources/META-INF/spring.factories (hunk 5 above)
Comment (alexjoybc) [reply src/backend/libs/efiling-demo-starter/src/main/resources/META-INF/spring.factories]: yup

File: src/backend/libs/efiling-lookup-client/src/test/java/ca/bc/gov/open/jag/efilinglookupclient/CSOLookupServiceImplTest.java (hunk 2 above)
Comment (alexjoybc) [reply src/backend/libs/efiling-lookup-client/src/test/java/ca/bc/gov/open/jag/efilinglookupclient/CSOLookupServiceImplTest.java:84]: nope

**Pull request merged**